From dfdc639736ac7b63492cfd29489426ad9b07f380 Mon Sep 17 00:00:00 2001
From: ran shalit <rshalit@habana.ai>
Date: Wed, 23 Feb 2022 10:38:14 +0200
Subject: [PATCH] [SW-75066] gaudi2: ppboot: fix calc num img slice

preboot image is divided to slice, each slice in size
of 64KB.
The calculation of number of slices missed an edge
case where the image size is exactly multiply number
of 64KB slices. This patch fix this calculation.

Change-Id: I38ff3c71129df4455e098b631130b30aec9216b7
Signed-off-by: ran shalit <rshalit@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index f1a2132..ae97ffd 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -313,7 +313,10 @@
 
 	flash_offset = ppboot_flash_image_offset(image_ptr);
 	image_size = spi_flash_readl(flash_offset);
-	num_cycles = (image_size/(MAX_DMA_SIZE + CRC_SIZE)) + 1;
+	num_cycles = (image_size / (MAX_DMA_SIZE + CRC_SIZE));
+	if (image_size % (MAX_DMA_SIZE + CRC_SIZE))
+		num_cycles++;
+
 	payload_size = image_size - (num_cycles * CRC_SIZE);
 
 	/*
